src/shogun/kernel/GaussianKernel.cpp

```diff
@@ -88,21 +89,40 @@ void CGaussianKernel::set_width(float64_t w)
 	m_log_width = std::log(w / 2.0) / 2.0;
 }
 
+auto CGaussianKernel::kernel_function(int32_t idx_a, int32_t idx_b)
+{
+	// this could be written as Eigen::Matrix<float64_t, n_differentiable_params, 1>;
+	m_eigen_log_width = Eigen::AutoDiffScalar<Eigen::VectorXd>(m_log_width);
```

Comment:
@karlnapf this should fix the temporary being deleted. Ideally we would replace m_log_width with this... Because this isn't thread safe

Reply:
and for the distance matrix we woudn't have this issue because features holds a reference to the data!